src/timeMachine/selectors/index.ts

```diff
@@ -293,14 +295,27 @@ export const getStartTime = (timeRange: TimeRange) => {
   }
 }
 
+// this function takes in a string that represents a date, and swaps the timezone to local.
+// if the date passed was 2021-09-02 12:00:00 UTC the function will return 2021-09-02 12:00:00 <Local TZ>
+const setTimeToLocal = (date?: string): Date => {
```

Comment:
i'm confused about the intent of this.

`valueOf` and `getTime` return the number of milliseconds and seconds in UTC. are you trying to return the number of milliseconds in local time?

Reply:
So, when we pass a string like `2019-09-02` into moment, it interprets the timezone as Local.
Whereas the Date package interprets it as UTC time zone. 

This makes a slight discrepancy in the `valueOf` return value. So to fix it, I change the timezone of the passed string to local. Just like the `[setTimeToUTC](https://github.com/influxdata/ui/blob/e198ecfb1341dbbeb2b8a2d9082d718919704810/src/dashboards/selectors/index.ts#L66)`

I found that out after reading this: https://stackoverflow.com/a/31402874

Reply:
>I'm trying to get the UTC timestamp in milliseconds but for local time

@ChitlangeSahas is the value we have passed in **always** going to be a string like `YYYY-MM-DD` ?